### PR TITLE
Provide option to anonymize node

### PR DESCRIPTION
**Pull request by chris-smith**
- Adds support for `anonymous` option when calling `initNode` to anonymize your node name. Follows rospy logic.
- Fixes error message when unable to find message package.
- Also some unintended whitespace changes? Maybe from Atom?

e.g.

```
rospy.initNode('/my_node', {anonymous: true})
.then((nh) => {
   console.log( nh.getNodeName() ); // "/my_node_25878_1464120320242"
});
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -81,19 +92,23 @@ let Rosnodejs = {
...
       if (nodeName === rosNode.getNodeName()) {
         return Promise.resolve(this.getNodeHandle());
       }
       // else
-      throw new Error('Unable to initialize node [' + nodeName + '] - node [' 
+      throw new Error('Unable to initialize node [' + nodeName + '] - node ['
                       + rosNode.getNodeName() + '] already exists');
     }
 
-    // FIXME: validate nodeName -- MUST START WITH '/'
```
Comment (IanTheEngineer) [inline index.js:95]: Isn't this FIXME still true? String validation of the node name hasn't changed.

Comment (chris-smith) [inline index.js:95]: I meant to remove it when I added the validation that exists. There is, admittedly, a lot more name validation/mapping that we need to do, though this exists for most ROS names and not specifically for node names.

File: utils/message_utils.js [hunk 2]
```diff
@@ -184,7 +184,7 @@ let MessageUtils = {
   loadMessagePackage(msgPackage) {
     const indexPath = messagePackagePathMap[msgPackage];
     if (indexPath === undefined) {
-      throw new Error('Unable to find message package %s', msgPackage);
+      throw new Error('Unable to find message package ' + msgPackage);
```
Comment (IanTheEngineer) [inline utils/message_utils.js:187]: Is there a JS style guide or linter dictating this change? If not, we should probably choose one. I assume this change is marginally better that it was previous?

Comment (chris-smith) [inline utils/message_utils.js:187]: error's won't format strings for you like logging will - so in the log you ended up just seeing
`Unable to find message package %s`

File: index.js [hunk 3]
```diff
@@ -81,19 +92,23 @@ let Rosnodejs = {
    * @return {Promise} resolved when connection to master is established
    */
   initNode(nodeName, options) {
+    options = options || {};
+    if (options.anonymous) {
+      nodeName = _anonymizeNodeName(nodeName);
+      console.log('anonymous node name ' + nodeName);
```
Comment (IanTheEngineer) [inline index.js]: Just an FYI, this is the only spot in the file we `console.log()`, or any logging at all for that matter. Will this be replaced by your `rosnodejs` logging in the future?

**Comment by IanTheEngineer** [comment]
Looks great overall. I'm ready to merge as is.

File: index.js (hunk 3 above)
Comment (chris-smith) [inline index.js]: whoops! testing - meant to pull this out

File: utils/message_utils.js (hunk 2 above)
Comment (IanTheEngineer) [inline utils/message_utils.js:187]: Noted! Not a marginal change at all, then.

File: index.js (hunk 3 above)
Comment (IanTheEngineer) [inline index.js]: It's not necessarily bad to have it in there, as the pid and time are not obvious to the user when starting the node.

Comment (chris-smith) [inline index.js]: Will people typically care that they see the node name when anonymizing though? `rospy` doesn't log anything. They can also ask for the name for the name of the node at any point.

```
console.log(rosnodejs.nh.getNodeName());
```

Comment (IanTheEngineer) [inline index.js]: True, that works. Let's remove it.

**Pull request merged**